cpp/src/arrow/c/dlpack.h

```diff
@@ -48,4 +51,7 @@ Result<DLManagedTensor*> ExportArray(const std::shared_ptr<Array>& arr);
 ARROW_EXPORT
 Result<DLDevice> ExportDevice(const std::shared_ptr<Array>& arr);
 
+ARROW_EXPORT
+Result<DLDevice> ExportTensorDevice(const std::shared_ptr<Tensor>& t);
```

Comment:
For the C++ side, it might be more natural API to call this `ExportDevice` as well (so just add an additional supported signature for the existing function)

Reply:
https://github.com/apache/arrow/pull/42118/commits/646b8b9d0ab4119618731329871076107f4bd2c1